include/pm/device.h

```diff
@@ -424,8 +431,59 @@ static inline bool pm_device_wakeup_is_capable(const struct device *dev)
 {
 	return false;
 }
+
 #endif /* CONFIG_PM_DEVICE */
 
+#ifdef CONFIG_PM_DEVICE_CONSTRAINT
+/**
+ * @brief Set a constraint on the current device state.
+ *
+ * This function sets a constraint in the current device state. After the
+ * constraint is set the system power management and device runtime power
+ * management will not longer change the device state until the constraint be
+ * released.
+ *
+ * @see pm_device_constraint_release
+ *
+ * @param dev Device instance.
+ */
+void pm_device_constraint_set(struct device *dev);
+
+/**
+ * @brief Release a constraint on the current device state.
+ *
+ * Releases a previously set constraint.
+ *
+ * @see pm_device_constraint_set
+ *
+ * @param dev Device instance.
+ */
+void pm_device_constraint_release(struct device *dev);
+
+/**
+ * @brief Check if the device pm has a constraint.
+ *
+ * @param dev Device instance.
+ *
+ * @retval true If device has a constraint.
+ * @retval false If device does not have a constraint.
+ */
+bool pm_device_constraint_check(const struct device *dev);
```

Comment:
`_is_set`?

Reply:
changed API namespace to `state_lock*`, so this function now is called `pm_device_state_is_locked`